freqtrade/freqai/data_kitchen.py

```diff
@@ -1126,13 +1126,57 @@ def set_all_pairs(self) -> None:
             if pair not in self.all_pairs:
                 self.all_pairs.append(pair)
 
+    def extract_corr_pair_columns_from_populated_indicators(
+        self,
+        dataframe: DataFrame
+    ) -> Dict[str, DataFrame]:
+        """
+        Find the columns of the dataframe corresponding to the corr_pairlist, save them
+        in a dictionary to be reused and attached to other pairs.
+        :params:
+        :dataframe: fully populated dataframe (current pair + corr_pairs)
+        :return:
+        :corr_dataframes: dictionary of dataframes to be attached to other pairs in same candle.
+        """
+        corr_dataframes: Dict[str, DataFrame] = {}
+        pairs = self.freqai_config["feature_parameters"].get("include_corr_pairlist", [])
+
+        for pair in pairs:
```

Comment:
I think we can improve performance by not looping through the pairs, both here and in `attach_corr_pair_columns`. 
Here we could simply do:
` corr_pairs = [pair.split('/')[0] for pair in pairs]
  corr_dataframes = dataframe.filter(regex='|'.join(corr_pairs))`
And in `attach_corr_pair_columns`:
`dataframe = pd.concat([dataframe, corr_dataframes], axis=1)
 if len(dataframe.columns.unique()) < len(dataframe.columns):
            dataframe = dataframe[[dataframe.columns.unique()]]`

Reply:
I'm, not sure the improvement in performance is worth the reduced readability here...we are talking about maybe miliseconds gained, but the proposed code is nearly impossible to come back to or for a new person to understand.